### ISSUE TITLE
Clarinet exits with `Killed` with too many tests

### ISSUE THREAD

**Issue by whoabuddy** [hirosystems/clarinet#324]
### Description

I'm rewriting a large number of our tests and noticed something new as I duplicated several versions of them.

When running `clarinet test` against a large number of test files and test cases (45 files, 534 unit tests) it exits with no error message and just the text `Killed.`

The number of files/tests will decrease a lot when we are done with the changes and we can work around this for now, but wanted to share in case this indicates a larger issue.

I caught a screenshot from top showing some pretty high usage right before it exited:

![Screenshot from 2022-04-20 11-00-09](https://user-images.githubusercontent.com/9038904/164294070-7a08d4d1-aacd-45c7-89c7-22b51abec268.png)

Sample output at the end:

```
* [NewYorkCityCoin Auth] JOB MANAGEMENT add-uint-argument() fails with ERR_UNKNOWN_JOB while adding argument to unknown job ID ... ok (63ms)
"/home/whoabuddy/Dev/citycoins/contracts/Clarinet.toml"
* [NewYorkCityCoin Auth] JOB MANAGEMENT add-uint-argument() fails with ERR_JOB_IS_ACTIVE while adding argument to active job ID ... ok (79ms)
"/home/whoabuddy/Dev/citycoins/contracts/Clarinet.toml"
* [NewYorkCityCoin Auth] JOB MANAGEMENT add-uint-argument() fails with ERR_UNAUTHORIZED while adding argument by someone who is not the job creator ... ok (57ms)
"/home/whoabuddy/Dev/citycoins/contracts/Clarinet.toml"
* [NewYorkCityCoin Core] MINING ACTIONS mine-many() succeeds and saves information that miner mined multiple consecutive blocks ... ok (145ms)
"/home/whoabuddy/Dev/citycoins/contracts/Clarinet.toml"
Killed
```

### To Reproduce

- Check out [the branch from](https://github.com/citycoins/contracts/tree/fix/test-mia-nyc) this PR: citycoins/contracts#187
- The version that produces this error is from commit https://github.com/citycoins/contracts/pull/187/commits/320125341d7651f9236587294ddcd92b2b7c84e6
- Run `clarinet test`

It also failed in the [related GitHub Action](https://github.com/citycoins/contracts/runs/6099961326?check_suite_focus=true).

### System Info

- Linux Mint 20.3 Cinnamon
- AMD Ryzen 9 5900X 12-Core Processor
- 16 GB RAM

**Comment by lgalabru** [comment]
Some good progress here. In https://github.com/hirosystems/clarinet/pull/333, I'm rewriting the test execution (based on the new deployment feature) and your feedbacks in mind.
When running the current version of Clarinet on https://github.com/citycoins/contracts/commit/320125341d7651f9236587294ddcd92b2b7c84e6,
I'm getting:
```bash
test result: ok. 575 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out (36794ms)
```
Memory usage: 16gb max

With this new version:
```bash
test result: ok. 575 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out (28651ms)
```
Memory usage: 500mb max

**Comment by lgalabru** [comment]
Addressed in #333
